lib/codeowners/cli/main.rb

```diff
@@ -61,25 +60,64 @@ def write_codeowners
       end
 
       def suggest_add_to_codeowners(file)
-        return unless yes?("File added: #{file.inspect}. Add owner to the CODEOWNERS file?")
+        return if @quit
 
-        owner = ask('File owner(s): ')
-        new_line = create_new_pattern(file, owner)
-
-        unless new_line.pattern?
-          puts "#{owner.inspect} is not a valid owner name."
-          return
+        case add_to_codeowners_dialog(file)
+        when 'y' then add_to_codeowners(file)
+        when 'i' then return
+        when 'q' then @quit = true
         end
+      end
+
+      def add_to_codeowners_dialog(file)
+        ask(<<~QUESTION, limited_to: %w[y i q])
+          File added: #{file.inspect}. Add owner to the CODEOWNERS file?
+          (y) yes
+          (i) ignore
+          (q) quit
+        QUESTION
+      end
+
+      def add_to_codeowners(file)
+        new_line = create_new_pattern(file)
 
         subgroups = @checker.main_group.subgroups_owned_by(new_line.owner)
         add_pattern(new_line, subgroups)
 
         @codeowners_changed = true
       end
 
-      def create_new_pattern(file, owner)
-        line = "#{file} #{owner}"
-        Codeowners::Checker::Group::Line.build(line)
+      def create_new_pattern(file)
+        list_owners
+        loop do
+          owner = new_owner
```

Comment:
it might be better to extract `sorted_owners` to the beginning of the method and pass it as a param:
```ruby
sorted_owners = @checker.main_group.owners.sort
list_owners sorted_owners
loop do
  owner = new_owner sorted_owners
```